pkg/rules/adapters/query.go

```diff
@@ -0,0 +1,52 @@
+package adapters
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/timescale/promscale/pkg/promql"
+)
+
+type queryAdapter struct {
+	queryable promql.Queryable
+}
+
+// NewQueryAdapter acts as an adapter to make Promscale's Queryable compatible with storage.Queryable
+func NewQueryAdapter(q promql.Queryable) *queryAdapter {
+	return &queryAdapter{q}
+}
+
+func (q *queryAdapter) Querier(ctx context.Context, mint int64, maxt int64) (storage.Querier, error) {
+	qr, err := q.queryable.SamplesQuerier(ctx, mint, maxt)
+	if err != nil {
+		return nil, fmt.Errorf("samples-querier: %w", err)
+	}
+	return querierAdapter{qr}, nil
+}
+
+type querierAdapter struct {
+	qr promql.SamplesQuerier
+}
+
+func (q querierAdapter) Select(sortSeries bool, hints *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
+	// Pushdowns are not supported here. This is fine as Prometheus rule-manager only uses queryable to know
+	// the previous state of the alert. This function is not used in recording/alerting rules evaluation.
+	seriesSet, _ := q.qr.Select(sortSeries, hints, nil, nil, matchers...)
+	return seriesSet
+}
+
+func (q querierAdapter) LabelValues(name string, _ ...*labels.Matcher) ([]string, storage.Warnings, error) {
+	// Weak TODO: We need to implement the matchers.
```

Comment:
is this function ever actually used? Does the lack of matching effect correctness?

Reply:
In alerting, no. I wrote todo because we need to update ourselves with prometheus, irrelevant of this PR

Reply:
I'd prefer this to be a panic if ever called then. I have significant concerns about what's implemented in terms of correctness